### PR TITLE
offroad/ui: Transparent layout mode and msgq-based layout control

### LINKED ISSUE #1293: UI bug when going onroad while in advanced wifi settings
screenshot

### PR DESCRIPTION
**Pull request by andyh2**
* Reimplements ui->offroad layout control, using `offroadLayout` socket instead of android broadcasts. 
* Adds UiLayoutState.App.None, with which offroad will draw a transparent/invisible activity window instead of its normal content.  
* Fixes regression in onboarding where the green engaged border did not appear after pressing the "SET" button on screen. It had previously relied upon Android broadcasts, so this bug appeared with the frame refactor. Now we communicate a field in UiLayoutState. 
* New APK (https://github.com/commaai/openpilot-apks/pull/47) supports the above changes and also resolves https://github.com/commaai/openpilot/issues/1293, https://github.com/commaai/openpilot/issues/1323.

### CODE REVIEW

**Comment by geohot** [comment]
Testing now

File: selfdrive/ui/ui.cc [hunk 1]
```diff
@@ -964,19 +1008,24 @@ int main(int argc, char* argv[]) {
       enable_event_processing(true);
       if (s->status != STATUS_STOPPED) {
         update_status(s, STATUS_STOPPED);
+        s->active_app = cereal_UiLayoutState_App_home;
+        update_offroad_layout_state(s);
```
Comment (geohot) [inline selfdrive/ui/ui.cc:1012]: I'm wondering if we should just send this every frame instead of on update.

**Comment by geohot** [comment]
Hmm, why does settings still seem to be slow when I press the button?

**Comment by geohot** [comment]
LGTM, nice job! Will merge this when the tests pass and fix up the bg thread PR

**Review by geohot — APPROVED** [review]

**Comment by andyh2** [comment]
Offroad is responsible for the remaining latency in opening settings. I'm looking into what the bottleneck is there. We should end up with at most the same perceived latency as pre-frame-refactor

**Comment by geohot** [comment]
Yea it actually seems a bit better.

**Pull request merged**